src/lib.rs

```diff
@@ -2788,6 +2802,9 @@ impl Build {
     }
 
     fn getenv(&self, v: &str) -> Option<String> {
+        if self.emit_rerun_if_env_changed {
```

Comment:
We only need to do this if the var was not cached, so this ideally would be only in the branch that fails to read from the cache.